core/pubnub_coreapi_ex.c

```diff
@@ -273,15 +273,22 @@ static enum pubnub_res heartbeat_with_state(pubnub_t *pb,
     if (rslt != PNR_OK) {
         return rslt;
     }
-
-    pb->core.state = state;
+    
+    size_t state_lenght = strlen(state);
```

Comment:
variable name typo? state_lenght...may be it should be state_length

Reply:
fixed